ui/src/app/settings/components/repos-list/repos-list.tsx

```diff
@@ -76,23 +75,101 @@ interface NewGitHubAppRepoCredsParams {
     tlsClientCertKey: string;
 }
 
-export class ReposList extends React.Component<RouteComponentProps<any>, {connecting: boolean}> {
+export enum ConnectionMethod {
+    SSH = 'via SSH',
+    HTTPS = 'via HTTPS',
+    GITHUBAPP = 'via GitHub App'
+}
+
+export class ReposList extends React.Component<
+    RouteComponentProps<any>,
+    {
+        connecting: boolean;
+        method: string;
+    }
+> {
     public static contextTypes = {
         router: PropTypes.object,
         apis: PropTypes.object,
         history: PropTypes.object
     };
 
-    private formApiSSH: FormApi;
-    private formApiHTTPS: FormApi;
-    private formApiGitHubApp: FormApi;
+    private formApi: FormApi;
     private credsTemplate: boolean;
     private repoLoader: DataLoader;
     private credsLoader: DataLoader;
 
     constructor(props: RouteComponentProps<any>) {
         super(props);
-        this.state = {connecting: false};
+        this.state = {
+            connecting: false,
+            method: ConnectionMethod.SSH
+        };
+    }
+
+    private ConnectRepoFormButton(method: string, onSelection: (method: string) => void) {
+        return (
+            <div className='white-box'>
+                <p>Choose your connection method:</p>
+                <DropDownMenu
+                    anchor={() => (
+                        <p>
+                            {method.toUpperCase()} <i className='fa fa-caret-down' />
+                        </p>
+                    )}
+                    items={[ConnectionMethod.SSH, ConnectionMethod.HTTPS, ConnectionMethod.GITHUBAPP].map(
+                        (type: ConnectionMethod.SSH | ConnectionMethod.HTTPS | ConnectionMethod.GITHUBAPP) => ({
+                            title: type.toUpperCase(),
+                            action: () => {
+                                onSelection(type);
+                                this.clearConnectRepoForm();
```

Comment:
This does clear the validation errors, but it also resets the form field values.

For example, if you had already entered the Repo URL, and then you switched the connection type, the field's value IMHO 'should' be retained.  With this change, the field values are cleared. 

Could you try just removing the validation errors and see if you like prefer this behavior?

So something like this:

```
                        const formState = this.formApi.getFormState();
                        this.formApi.setFormState({
                            ...formState,
                            errors: {}
                        });
```

Reply:
sure, I like that better too. Thank you!